packages/hardhat-core/src/internal/hardhat-network/provider/modules/debug.ts

```diff
@@ -0,0 +1,43 @@
+import { rpcHash } from "../../../core/jsonrpc/types/base-types";
+import {
+  rpcDebugTracingConfig,
+  RpcDebugTracingConfig,
+} from "../../../core/jsonrpc/types/input/debugTraceTransaction";
+import { validateParams } from "../../../core/jsonrpc/types/input/validation";
+import { MethodNotFoundError } from "../../../core/providers/errors";
+import { HardhatNode } from "../node";
+
+// tslint:disable only-hardhat-error
+
+export class DebugModule {
+  constructor(private readonly _node: HardhatNode) {}
+
+  public async processRequest(
+    method: string,
+    params: any[] = []
+  ): Promise<any> {
+    switch (method) {
+      case "debug_traceTransaction":
+        return this._traceTransactionAction(
+          ...this._traceTransactionParams(params)
+        );
+    }
+
+    throw new MethodNotFoundError(`Method ${method} not found`);
+  }
+
+  // debug_traceTransaction
+
+  private _traceTransactionParams(
+    params: any[]
+  ): [Buffer, RpcDebugTracingConfig] {
+    return validateParams(params, rpcHash, rpcDebugTracingConfig);
+  }
+
+  private async _traceTransactionAction(
+    hash: Buffer,
+    config: RpcDebugTracingConfig
+  ): Promise<object> {
```

Comment:
Why is this `object` instead of `RpcDebugTraceOutput`? I suspect this is here because `HardhatNode.traceTransaction` returns `Promise<object>`, but if its return type is removed ts correctly infers `Promise<RpcDebugTraceOutput>`.

Reply:
I have no idea why I did this. Fixed in 6822de29e99e6ab617aa2b5b2bfc7a009beda637.